source/horoscope/js/horoscope.js

```diff
@@ -25,28 +25,33 @@ async function init() {
     }
 
     // get and load in the fortune as well as the sign
-
-    fortuneElement.innerText = await getPrompt();
-    fortuneElementTitle.innerText = dateToHoroscope(date);
+    if (fortuneElement) {
+        fortuneElement.innerText = await getPrompt();
+        fortuneElementTitle.innerText = dateToHoroscope(date);
+    }
 
     // Add event listener for the save button
-    save.addEventListener('click', async () => {
-        let bday = localStorage.getItem('birthday');
-        let sign = dateToHoroscope(bday);
-        let category = localStorage.getItem('category');
-        let message = fortuneElement.innerText;
-        let today = new Date();
-        
-        //save horoscope to local storage for sidebar
-        let horoscopeElement = new Horoscope(sign, bday, today, message, category);
-        saveHoroscope(horoscopeElement);
-        window.location.href = HISTORY_PAGE;
-    })
+    if (save) {
```

Comment:
why do you need to check for the existence of save?
is it because there is a case where the Save Element is still rendering when this is being called?

Reply:
horoscope.js is also pulled on by the landing page but those elements only exist on the horoscope.html page. if you don't check for the existence of save the page works but there are console errors where we are trying to read or access attributes of null elements. 

Checks is just to decrease the number of console errors we get.

Reply:
i see. gotcha.